ingestion/tests/unit/test_source_connection.py

```diff
@@ -209,3 +241,365 @@ def test_pinotdb_url(self):
             pinotControllerHost="http://localhost:9000/",
         )
         assert expected_url == get_connection_url(pinot_conn_obj)
+
+    def test_mysql_url(self):
+        # connection arguments without db
+        expected_url = "mysql+pymysql://openmetadata_user:@localhost:3306"
+        mysql_conn_obj = MysqlConnection(
+            username="openmetadata_user",
+            hostPort="localhost:3306",
+            scheme=MySQLScheme.mysql_pymysql,
+            database=None,
+        )
+        assert expected_url == get_connection_url(mysql_conn_obj)
+
+        # connection arguments with db
+        expected_url = "mysql+pymysql://openmetadata_user:@localhost:3306/default"
+        mysql_conn_obj = MysqlConnection(
+            username="openmetadata_user",
+            hostPort="localhost:3306",
+            scheme=MySQLScheme.mysql_pymysql,
+            database="default",
+        )
+        assert expected_url == get_connection_url(mysql_conn_obj)
+
+    def test_clickhouse_url(self):
+        # connection arguments without db
+        expected_url = "clickhouse+http://username:@localhost:8123"
+        clickhouse_conn_obj = ClickhouseConnection(
+            username="username",
+            hostPort="localhost:8123",
+            scheme=ClickhouseScheme.clickhouse_http,
+            database=None,
+        )
+        assert expected_url == get_connection_url(clickhouse_conn_obj)
```

Comment:
Does it work without password?
are we able to establish connection?

Reply:
@ayush-shah yes it's work without password.